File: src/liberary/account_lib/account/pub_account_info.rs
use serde::{Deserialize, Serialize};

use crate::liberary::account_lib::{account::{account_customisation::AccountCustomisation, user_data::{radar_chart::RadarChart, shiftstone::Shiftstone, account_stats::AccountStats}}, schedule::schedule::Schedule};
use super::account::Account;

#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct PubAccountInfo {
    pub id: String,
    pub username: String,
    pub stats: AccountStats,
    pub avatar: Option<String>,
    pub schedule: Option<Schedule>,
    pub customisation: Option<AccountCustomisation>,
}

impl Account {

    pub fn get_pub_info(&self) -> PubAccountInfo {
        PubAccountInfo {
            id: self.user_info.id.clone(),
            username: self.user_info.username.clone(),
            stats: self.stats.clone(),
            avatar:  self.user_info.avatar.clone(),
            schedule: self.schedule.clone(),
            customisation: self.customisation.clone(),
        }
    }
}

File: PORC-Front/src/models/pub_account_info/PubAccountInfo.ts
import type { AccountCustomisation } from "./account_cust/AccountCustomisation";
import { scheduleToRecv, type Schedule } from "../schedule/Schedule";
import type { PubAccountInfoRecv } from "./PubAccountInfoRecv";
import type { AccountStats } from "./account_stats/AccountStats";

export interface PubAccountInfo {
    id: string;
    username: string;
    avatar: string | null;
    schedule: Schedule | null;
    customisation: AccountCustomisation | null;
    stats: AccountStats;
}

export function pubAccountInfoToRecv(val: PubAccountInfo): PubAccountInfoRecv {
    return {
        id: val.id,
        username: val.username,
        avatar: val.avatar,
        customisation: val.customisation,
        stats: val.stats,
        schedule: val.schedule === null
            ? null
            : scheduleToRecv(val.schedule),
    } as PubAccountInfoRecv
}

